server/controllers/profile.js

```diff
@@ -0,0 +1,102 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/create
+// @desc create a profile
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const {
+        userId,
```

Comment:
You can get the `userId` from `req.user.id` then check validity with the User database. Getting this from the request body risks profile creation by another user.